contracts/p1/mixins/Trading.sol

```diff
@@ -28,10 +31,24 @@ abstract contract TradingP1 is Multicall, ComponentP1, ReentrancyGuardUpgradeabl
     uint48 public tradesOpen;
 
     // === Governance param ===
-    uint192 public maxTradeSlippage; // {%}
+    uint192 public maxTradeSlippage; // {1}
 
     uint192 public minTradeVolume; // {UoA}
 
+    // === Added in 3.0.0 ===
+
+    // {s} the length of the implicit falling-price dutch auction
+    uint48 public dutchAuctionLength;
+
+    // At the start of a tx, tradeEnd can be:
+    //   1. more than dutchAuctionLength away => No dutch auction ongoing
+    //   2. within dutchAuctionLength in the past => Dutch auction with 0 bids ongoing
+    //   3. within dutchAuctionLength in the future => Dutch auction with 1+ bids ongoing
+    // [X, Y): inclusive on the left-bound and exclusive on the right-bound
+    uint48 internal tradeEnd; // {s} timestamp of the end of the last trade (batch OR dutch)
```

Comment:
i think we want to clean up the usage of the word "trade". on line 225 we distinguish that a dutch-auction swap is not a trade https://github.com/reserve-protocol/protocol/pull/777/files#diff-b3b12b74962f08fc087a7944168db17cebbd1f7c989fe73a7b047957e6a450d4R225 but we also seem to be using this `tradeEnd` variable to time a dutch-auction swap.  maybe `lastAuctionEnd` is a better name here?

Reply:
Yeah.....good point.

Reply:
Will think on this. The other possible direction is to rename `tradesOpen` to something more narrow, like `nonAtomicTradesOpen` or something